spec/controllers/api/v1/books_controller_spec.rb

```diff
@@ -0,0 +1,129 @@
+require 'rails_helper'
+
+shared_context 'Authenticated User' do
+  let(:user) { create(:user) }
+
+  before do
+    request.headers.merge! user.create_new_auth_token
+  end
+end
+
+describe API::V1::BooksController, type: :controller do
+  include_context 'Authenticated User'
+
+  describe 'GET #index' do
+    context 'When fetching all the books' do
+      let!(:books) { create_list(:book, 3) }
```

Comment:
Do not Use let! for test setup: https://www.rubydoc.info/gems/rubocop-rspec/1.7.0/RuboCop/Cop/RSpec/LetSetup